dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -388,6 +388,18 @@ class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink, public virtual
     mutable ACE_Thread_Mutex mutex_;
     mutable ACE_Thread_Mutex elems_not_acked_mutex_;
 
+    typedef PmfSporadicTask<RtpsWriter> Sporadic;
+
+    mutable ACE_Thread_Mutex heartbeat_mutex_;
```

Comment:
Are you sure need heartbeat_mutex_ and can't just use mutex_? The only reason elems_not_acked_mutex_ exists is because of some weird reentrant behavior with the send buffer, I'm pretty sure. I wouldn't have expected the heartbeat stuff to need something similar if it didn't need it before.